pkg/windows/windows.go

```diff
@@ -419,15 +419,27 @@ func (vm *windows) Deconfigure(watchNamespace string, apiServerURL string) error
 	return nil
 }
 
-func (vm *windows) Bootstrap(desiredVer, apiServerURL, watchNamespace string, credentials *Authentication) error {
+func (vm *windows) Bootstrap(desiredVer, apiServerURL, watchNamespace string, credentials *Authentication, nodeExists bool) error {
 	vm.log.Info("configuring")
-	// Make sure WICD service is not running before calling node bootstrap
-	if err := vm.deconfigureWICD(); err != nil {
+
+	// Make sure WICD service is not running before calling node cleanup and/or bootstrap
+	if err := vm.ensureServiceIsRemoved(WicdServiceName); err != nil {
+		return errors.Wrapf(err, "error ensuring %s Windows service is removed", WicdServiceName)
+	}
+	// We have to do ensure WICD exists separately before the rest of the files because we must ensure services are
+	// stopped and their files closed before modifying them.
+	if err := vm.ensureWICDExists(); err != nil {
 		return err
 	}
-	if err := vm.ensureRequiredServicesStopped(); err != nil {
-		return errors.Wrap(err, "unable to stop all services")
+	// Stop any services that may be running. This prevents the node being shown as Ready after a failed configuration.
+	// TODO: WICD cleanup can only be called if a node object exists since it relies on the desired version annotation.
+	// The nodeExists flag should be removed once best-effort cleanup is done, https://issues.redhat.com/browse/WINC-853
+	if nodeExists {
```

Comment:
Shouldn't you stopping services always even if the node object does not exist like we were doing before? Consider the case where we error out during kubelet setup and should start again. At that point it is best we stop all the services kubelet depends on.

Reply:
This is the issue that almost made me drop this from the story's acceptance criteria. WICD [does not support cleanup if a node object doesn't exist](https://github.com/openshift/windows-machine-config-operator/blob/master/pkg/daemon/cleanup/cleanup.go#L53) - this is part of the best-effort cleanup work that was punted from the epic.

Reply:
One way around this is to pass the desiredVersion to the cleanup command instead of relying on a node annotation, as we do to the bootstrap command. @sebsoto likely has thoughts on this, so looping him in

Reply:
It does seem weird that WICD cannot perform cleanup if the node object is not present.

>One way around this is to pass the desiredVersion to the cleanup command instead of relying on a node annotation, as we do to the bootstrap command.

I am fine with that. Why do we even need the `desiredVersion` annotation if we are passing it to WICD?

Reply:
> Why do we even need the desiredVersion annotation if we are passing it to WICD

The original thought process there is that by changing the desiredVersion annotation we can trigger an upgrade without stopping and changing the WICD service, in the case where there is no change to the WICD exe.

>  One way around this is to pass the desiredVersion to the cleanup command instead of relying on a node annotation, as we do to the bootstrap command.

This works for me. Another way is to just use the latest windows-services configmap in the absence of a node object

Reply:
> Another way is to just use the latest windows-services configmap in the absence of a node object

I went ahead and implemented this (which is half of https://issues.redhat.com/browse/WINC-853)